In [176]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


배수펌프 = pd.read_csv('./데이터/Green/서울시 배수펌프장 정보.csv',encoding='cp949')
녹지 = pd.read_csv('./데이터/Green/서울시_녹지현황.csv',encoding='utf-8')
습윤지수 = pd.read_csv('./데이터/Green/서울시_습윤지수_병합.csv',encoding='cp949')
불투수율 = pd.read_csv('./데이터/Green/서울특별시_자치구별 불투수면적 현황.csv',encoding='cp949')
침수흔적도 = pd.read_csv('./데이터/서울시_침수흔적도.csv',encoding='utf-8')
빗물펌프거리= pd.read_csv('./데이터/서울시_빗물펌프거리.csv',encoding='utf-8')
하천거리 = pd.read_csv('./데이터/서울시_빗물펌프거리.csv',encoding='utf-8')

# 배수펌프
- 시설 관리자 구형식 : 시설관리자 => 지역구로 변경
- 표고 관련 column : null값 너무 많아서 뺌
- 전양정,실양정 : null값 너무 많아서 뺌
전양정,실양정 관련 글 : https://blog.naver.com/PostView.nhn?blogId=hvackkw&logNo=220875366802
- 지역구 : 서울시 => 구로구 인거 확인, 나머지도 %구 형식으로 변경

In [177]:
# 시설관리자가 구형식으로 되어있어서 null값에 구단위로 추가 후 구단위로 변경
배수펌프.loc[배수펌프['시설관리자'].isnull(), '시설관리자'] = 배수펌프['상세주소'].str.extract(r'(\S+구)')[0]

배수펌프 = 배수펌프.rename(columns = {'시설관리자': '자치구'})

In [178]:
# 자치구 조정 함수
def adjust_district_names(df):
    # '서울시'를 '구로구'로 변경
    df['자치구'] = df['자치구'].replace('서울시', '구로구')
    
    # '구청'을 '구'로 변경
    df['자치구'] = df['자치구'].str.replace('구청', '구')

    # '구구'가 되는 경우를 '구'로 변경
    df['자치구'] = df['자치구'].str.replace('구구', '구')
    
    # 모든 지역구가 '구'로 끝나도록 조정
    df['자치구'] = df['자치구'].apply(lambda x: x if x.endswith('구') else x.split()[0] + '구')
    
    return df

# 조정된 지역구 데이터프레임
배수펌프 = adjust_district_names(배수펌프)

In [179]:
배수펌프['상세주소']

0      서울시 구로구 구로4동 120-5
1                 풍납동 471
2               방이동 88-13
3               방이동 88-13
4               방이동 88-13
             ...         
91       서울시 성동구 송정동 78-1
92      서울시 동대문구 송정동 78-1
93       서울시 동대문구 장안동 356
94    서울시 동대문구 장안4동 301-4
95      서울시 동대문구 장안1동 379
Name: 상세주소, Length: 96, dtype: object

In [180]:
배수펌프.isna().sum()

하천기본계획코드       0
일련번호           0
하천기본계획 사업명     0
수립년도           0
시설물명           0
하천명            0
하천등급           0
상세주소           0
주소            68
유역면적           1
보호면적          10
시행기관          86
본류하천_유역면적      5
본류하천_홍수량       9
본류하천_계획빈도      9
제방고            9
계획홍수위          3
부지표고          61
기계실바닥표고       62
전기실바닥표고       87
배수장_최대배수량      1
배수장_계획빈도       1
초기흡입수위        24
전동기            0
펌프규격          29
전양정           58
실양정           91
토출수위          41
자치구            0
구분            22
배수문규격         33
배수문바닥표고       42
간선수로          89
지선수로          89
유수지표고_HWL     95
유수지표고_LWL     86
유수지면적         61
유수지용량         42
dtype: int64

In [181]:
배수펌프 = 배수펌프[['자치구','상세주소','시설물명','유역면적','보호면적','본류하천_유역면적','본류하천_홍수량','제방고','계획홍수위','배수장_최대배수량','초기흡입수위','펌프규격','토출수위','배수문규격','유수지면적','유수지용량']]

# 녹지
- 세부지역 => 지역구로 변경
- 소계 => 사용 x  삭제
- 구단위 => 중부공원 녹지사업소, 안전총괄실=> 중구로 합치기

In [182]:
# '소계'인 행을 제거
녹지 = 녹지[녹지['세부지역'] != '소계']


# NaN 값을 0으로 대체
녹지.fillna(0, inplace=True)
# '중구' 행을 찾기
중구_행 = 녹지[녹지['세부지역'] == '중구']

# '중부공원녹지사업소'와 '안전총괄실' 행을 찾기
중부공원_녹지사업소_행 = 녹지[녹지['세부지역'] == '중부공원녹지사업소']
안전총괄실_행 = 녹지[녹지['세부지역'] == '안전총괄실']

# 수치 데이터 컬럼만 선택하여 중구 행에 합산
수치_컬럼 = 녹지.columns[1:]  # 나머지 컬럼도 추가
중구_행[수치_컬럼] += 중부공원_녹지사업소_행[수치_컬럼].iloc[0] + 안전총괄실_행[수치_컬럼].iloc[0]

# '중부공원녹지사업소'와 '안전총괄실' 행을 데이터프레임에서 제거
녹지 = 녹지[~녹지['세부지역'].isin(['중부공원녹지사업소', '안전총괄실'])]

# 변경된 '중구' 행을 원본 데이터프레임에 업데이트
녹지.loc[녹지['세부지역'] == '중구', 수치_컬럼] = 중구_행[수치_컬럼]

녹지['세부지역'] = 녹지['세부지역'].replace('중구중부공원녹지사업소안전총괄실', '중구')

녹지 = 녹지.rename(columns = {'세부지역': '자치구'})


# '지역'컬럼 삭제
녹지= 녹지.drop(columns=['지역'])

C:\Users\ciw96\AppData\Local\Temp\ipykernel_27048\3469685596.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  중구_행[수치_컬럼] += 중부공원_녹지사업소_행[수치_컬럼].iloc[0] + 안전총괄실_행[수치_컬럼].iloc[0]
C:\Users\ciw96\AppData\Local\Temp\ipykernel_27048\3469685596.py:22: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[414.0]' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  녹지.loc[녹지['세부지역'] == '중구', 수치_컬럼] = 중구_행[수치_컬럼]
C:\Users\ciw96\AppData\Local\Temp\ipykernel_27048\3469685596.py:22: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[1944645.0]' has dtype incompati

In [183]:
녹지.columns

Index(['자치구', '총개소수', '총면적', '시설녹지_개소수', '시설녹지_면적', '일반녹지_개소수', '일반녹지_면적',
       '중앙분리대_개소수', '중앙분리대_면적', '교통섬_개소수', '교통섬_면적', '화단_개소수', '화단_면적',
       '보도부_개소수', '보도부_면적', '기타녹지_개소수', '기타녹지_면적', '지하철환기구_개소수', '지하철환기구_면적',
       '건물주변_개소수', '건물주변_면적', '아파트학교_개소수', '아파트학교_면적', '침수공간_개소수', '침수공간_면적',
       '기타_개소수', '기타_면적', '녹지율'],
      dtype='object')

In [184]:
습윤지수 = 습윤지수.rename(columns={'구별코드' :'자치구'})

In [185]:
습윤지수

자치구   TWI_min    TWI_max  TWI_count  \
0    강남구  0.956859  22.990770        256   
1    강동구  0.795508  26.145687        256   
2    강북구  0.442591  22.733292        256   
3    강서구  0.588825  26.614878        256   
4    관악구  0.451903  23.116009        256   
5    광진구  0.631392  20.507568        256   
6    구로구  0.939221  22.691904        256   
7    금천구  0.605828  24.525225        256   
8    노원구  0.615565  25.458551        256   
9    도봉구  0.483903  24.166096        256   
10  동대문구  0.656473  23.542290        256   
11   동작구  0.586479  23.116009        256   
12   마포구  0.692516  22.138985        256   
13  서대문구  0.653973  22.877716        256   
14   서초구  0.542244  28.288298        256   
15   성동구  0.822647  23.542290        256   
16   성북구  0.641562  27.020590        256   
17   송파구  0.887464  28.288298        256   
18   양천구  0.902545  23.466768        256   
19  영등포구  0.823762  23.592905        256   
20   용산구  0.591247  22.791409        256   
21   은평구  0.568285  23.601517        256   
22   종로구  0.406663  22.952368        256   
23    중구  0.660128  21.997917        256   
24   중랑구  0.680409  24.132950        256   

                                           TWI_counts  
0   10|21|67|148|218|374|576|934|1193|1619|1997|25...  
1   1|3|4|13|28|75|140|226|405|613|879|1221|1653|1...  
2   1|3|19|66|119|252|465|743|1279|1843|2439|3192|...  
3   5|3|30|41|78|191|372|644|934|1214|1483|1816|20...  
4   3|9|41|107|178|362|626|1084|1671|2653|3515|439...  
5   4|9|9|34|49|89|155|302|381|514|587|744|898|949...  
6   6|7|28|48|109|158|319|517|737|948|1363|1734|19...  
7   5|5|10|20|43|70|163|364|511|755|986|1386|1679|...  
8   5|16|23|85|171|324|591|994|1542|2257|2912|3590...  
9   6|14|34|56|95|144|180|296|433|587|755|1134|133...  
10  12|14|20|23|29|44|81|117|174|259|310|411|531|6...  
11  3|6|17|22|54|81|162|238|384|516|776|941|1061|1...  
12  9|18|33|75|151|291|451|701|1026|1381|1810|2167...  
13  13|36|83|174|332|551|886|1262|1644|2127|2723|3...  
14  3|5|17|57|173|439|893|1727|2765|4223|5812|7801...  
15  6|9|26|36|57|88|133|176|213|323|344|451|589|74...  
16  6|43|118|310|646|1251|1870|2568|3317|4004|4691...  
17  7|13|39|111|223|479|762|1183|1453|2014|2584|31...  
18  4|9|27|46|108|211|336|513|733|859|1110|1291|15...  
19  2|10|6|23|47|92|101|178|250|345|476|551|662|98...  
20  4|4|6|26|48|88|128|248|366|437|611|770|970|112...  
21  14|40|105|285|557|1095|1907|3024|4422|5457|665...  
22  5|8|37|118|190|420|776|1336|2244|3135|4092|479...  
23  3|7|13|9|52|63|139|207|298|397|523|664|839|935...  
24  4|4|15|33|92|140|262|384|562|784|934|1128|1359...

# 불투수율 

- 비고 삭제

In [186]:
불투수율 = 불투수율.drop(columns=['비고','연번'])

# 침수흔적도 

- 침수 시작 날짜 : 218로 시작=> 2018로 바꿈
- 침수 종료 날짜 : 틀 맞지 않는 형식 %Y%m%d형식으로 변경
-          추가  : Nan값들 침수시작날짜와 동일하게 변경

In [187]:
print(len(침수흔적도))
침수흔적도 = 침수흔적도[(침수흔적도['침수시작날짜']>=20180101) & (침수흔적도['침수시작날짜']<=20231231)]
침수흔적도
print(len(침수흔적도))

9724
9681


In [188]:
#
침수흔적도['침수시작날짜'] = 침수흔적도['침수시작날짜'].replace(2180828, 20180828)

# '침수종료날짜' 컬럼의 날짜 형식 통일
침수흔적도['침수종료날짜'] = pd.to_datetime(침수흔적도['침수종료날짜'], errors='coerce').dt.strftime('%Y%m%d')


# 'NaN'을 실제 NaN 값으로 변환
침수흔적도['침수종료날짜'] = 침수흔적도['침수종료날짜'].replace('NaN', np.nan)

# '침수종료날짜'의 NaN 값을 '침수시작날짜'로 대체
침수흔적도['침수종료날짜'] = pd.to_datetime(침수흔적도['침수종료날짜'], errors='coerce').dt.strftime('%Y%m%d')
침수흔적도['침수종료날짜'] = 침수흔적도['침수종료날짜'].fillna(침수흔적도['침수시작날짜'].astype(str))


In [189]:
# 문자열로 저장된 날짜를 정수형으로 변환
침수흔적도['침수시작날짜'] = 침수흔적도['침수시작날짜'].astype(int)
침수흔적도['침수종료날짜'] = 침수흔적도['침수종료날짜'].astype(int)

check = 0

# 각 행에 대해 침수종료날짜와 침수시작날짜의 차이를 계산하여 누적
for index, row in 침수흔적도.iterrows():
    check += row['침수종료날짜'] - row['침수시작날짜']


print(check)

70066


In [190]:
# 침수날짜 컬럼 생성: 각 행에 대해 침수 시작부터 종료까지의 모든 날짜 생성
침수흔적도['침수날짜'] = 침수흔적도.apply(
    lambda x: pd.date_range(
        start=pd.to_datetime(str(x['침수시작날짜']), format='%Y%m%d'),  
        end=pd.to_datetime(str(x['침수종료날짜']), format='%Y%m%d'),    
        freq='D'
    ).strftime('%Y%m%d').tolist(),
    axis=1
)

# 기존 '침수시작날짜', '침수종료날짜' 컬럼 제거
침수흔적도.drop(['침수시작날짜', '침수종료날짜'], axis=1, inplace=True)

# explode를 사용하여 '침수날짜'의 각 날짜를 독립적인 행으로 확장
침수흔적도 = 침수흔적도.explode('침수날짜')

print(침수흔적도)



       자치구  침수수심               침수원인  침수평균해발고도      침수면적        행정코드  침수시작시간  \
0      강북구  0.02  2022년 8.8 ~17. 호우    114.53   3876.00  1130510400      18   
1      강북구  0.02  2022년 8.8 ~17. 호우    118.05   2188.00  1130510400      18   
2      강북구  0.02  2022년 8.8 ~17. 호우    129.55  45287.00  1130510400      18   
3     영등포구  0.30  2022년 8.8 ~17. 호우     10.18     53.00  1156010400      20   
3     영등포구  0.30  2022년 8.8 ~17. 호우     10.18     53.00  1156010400      20   
...    ...   ...                ...       ...       ...         ...     ...   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   

      침수종료시간  침수내용                      주소  침수연도 침수

In [191]:
침수흔적도

자치구  침수수심               침수원인  침수평균해발고도      침수면적        행정코드  침수시작시간  \
0      강북구  0.02  2022년 8.8 ~17. 호우    114.53   3876.00  1130510400      18   
1      강북구  0.02  2022년 8.8 ~17. 호우    118.05   2188.00  1130510400      18   
2      강북구  0.02  2022년 8.8 ~17. 호우    129.55  45287.00  1130510400      18   
3     영등포구  0.30  2022년 8.8 ~17. 호우     10.18     53.00  1156010400      20   
3     영등포구  0.30  2022년 8.8 ~17. 호우     10.18     53.00  1156010400      20   
...    ...   ...                ...       ...       ...         ...     ...   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   

      침수종료시간  침수내용                      주소  침수연도 침수지역정보  \
0         23  내수침수     서울특별시 강북구 우이동 296-1  2022     주택   
1         23  내수침수     서울특별시 강북구 우이동 299-2  2022     주택   
2         23  내수침수     서울특별시 강북구 우이동 산8-15  2022     주택   
3         24  내수침수  서울특별시 영등포구 영등포동3가 13-7  2022     상가   
3         24  내수침수  서울특별시 영등포구 영등포동3가 13-7  2022     상가   
...      ...   ...                     ...   ...    ...   
9681      24  내수침수            양천구 신월동 97-4  2018    NaN   
9681      24  내수침수            양천구 신월동 97-4  2018    NaN   
9681      24  내수침수            양천구 신월동 97-4  2018    NaN   
9681      24  내수침수            양천구 신월동 97-4  2018    NaN   
9681      24  내수침수            양천구 신월동 97-4  2018    NaN   

                                                  지오메트리      침수날짜  
0     POLYGON ((956425.1271956386 1963792.043624141,...  20220808  
1     POLYGON ((956590.6408143371 1963700.11134354, ...  20220808  
2     POLYGON ((956383.2795007367 1963392.863235048,...  20220808  
3     POLYGON ((947842.214000747 1946702.945905377, ...  20220808  
3     POLYGON ((947842.214000747 1946702.945905377, ...  20220809  
...                                                 ...       ...  
9681  POLYGON ((940810.003042229 1948700.383580515, ...  20180828  
9681  POLYGON ((940810.003042229 1948700.383580515, ...  20180829  
9681  POLYGON ((940810.003042229 1948700.383580515, ...  20180830  
9681  POLYGON ((940810.003042229 1948700.383580515, ...  20180831  
9681  POLYGON ((940810.003042229 1948700.383580515, ...  20180901  

[33649 rows x 14 columns]

In [192]:
빗물펌프거리  = 빗물펌프거리.rename(columns={'구단위':'자치구'})

In [193]:
하천거리 = 하천거리.rename(columns={'구단위' : '자치구'})

In [194]:
침수흔적도['지오메트리'][0]

'POLYGON ((956425.1271956386 1963792.043624141, 956424.8854296223 1963792.446760627, 956421.6884048309 1963796.543246887, 956418.2632162417 1963800.405033851, 956416.8117928583 1963801.728263483, 956416.4236773055 1963801.303485454, 956415.673762397 1963802.019208943, 956413.8817620513 1963803.398229473, 956420.995892275 1963802.790613666, 956421.7707828018 1963809.404039349, 956425.7187906543 1963815.696705643, 956425.8033045963 1963815.799218082, 956434.6548788747 1963826.488171006, 956435.3412276129 1963823.593593678, 956436.8860344886 1963786.817015585, 956437.7349660392 1963768.627245897, 956495.3452938531 1963810.305350579, 956524.1096722074 1963776.098037665, 956516.0908221444 1963772.371073411, 956522.9037674818 1963753.847705869, 956517.5043732296 1963749.449057447, 956515.3938606675 1963740.933442291, 956502.5895697867 1963742.429994271, 956487.0221593921 1963745.967486548, 956462.8295554036 1963753.56735047, 956460.856231007 1963766.90490005, 956433.1606194369 1963759.031136

In [195]:
from functools import reduce

dfs = [녹지,습윤지수,불투수율,침수흔적도]

df = reduce(lambda left, right:pd.merge(left,right, on='자치구',how='outer'),dfs)


df

자치구   총개소수        총면적 시설녹지_개소수   시설녹지_면적 일반녹지_개소수   일반녹지_면적 중앙분리대_개소수  \
0      강남구  423.0  2002549.5     94.0  411351.8    116.0  179994.0       7.0   
1      강남구  423.0  2002549.5     94.0  411351.8    116.0  179994.0       7.0   
2      강남구  423.0  2002549.5     94.0  411351.8    116.0  179994.0       7.0   
3      강남구  423.0  2002549.5     94.0  411351.8    116.0  179994.0       7.0   
4      강남구  423.0  2002549.5     94.0  411351.8    116.0  179994.0       7.0   
...    ...    ...        ...      ...       ...      ...       ...       ...   
33645  중랑구  294.0   539974.0     31.0  128636.0    129.0   92593.0       0.0   
33646  중랑구  294.0   539974.0     31.0  128636.0    129.0   92593.0       0.0   
33647  중랑구  294.0   539974.0     31.0  128636.0    129.0   92593.0       0.0   
33648  중랑구  294.0   539974.0     31.0  128636.0    129.0   92593.0       0.0   
33649  중랑구  294.0   539974.0     31.0  128636.0    129.0   92593.0       0.0   

      중앙분리대_면적 교통섬_개소수  ...      침수면적          행정코드 침수시작시간 침수종료시간  침수내용  \
0      32057.8     7.0  ...  44306.00  1.168010e+09   20.0   23.0  내수침수   
1      32057.8     7.0  ...   2356.73  1.168010e+09   20.0   23.0  내수침수   
2      32057.8     7.0  ...   2015.54  1.168010e+09   20.0   23.0  내수침수   
3      32057.8     7.0  ...   2177.61  1.168010e+09   20.0   23.0  내수침수   
4      32057.8     7.0  ...      3.01  1.168010e+09   20.0   23.0  내수침수   
...        ...     ...  ...       ...           ...    ...    ...   ...   
33645      0.0     3.0  ...     12.00  1.126010e+09   10.0   12.0  내수침수   
33646      0.0     3.0  ...     40.22  1.126011e+09    9.0   10.0  내수침수   
33647      0.0     3.0  ...     10.00  1.126011e+09   14.0   16.0  내수침수   
33648      0.0     3.0  ...     73.79  1.126010e+09    9.0   12.0  내수침수   
33649      0.0     3.0  ...     20.00  1.126011e+09   17.0   18.0  내수침수   

                                주소    침수연도 침수지역정보  \
0                서울특별시 강남구 역삼동 858  2022.0     도로   
1             서울특별시 강남구 개포동 산53-38  2022.0     주택   
2                서울특별시 강남구 개포동 570  2022.0     주택   
3              서울특별시 강남구 개포동 571-1  2022.0     주택   
4      서울특별시 강남구 개포동 598-5 6-40-23  2022.0     주택   
...                            ...     ...    ...   
33645                         중화지구  2022.0     주택   
33646                         신내지구  2022.0     주택   
33647                         신내지구  2022.0     주택   
33648                         묵동지구  2022.0     주택   
33649                         신내지구  2022.0     주택   

                                                   지오메트리      침수날짜  
0      POLYGON ((958266.7548537395 1944412.821693935,...  20220808  
1      POLYGON ((961656.4046361586 1941847.951360239,...  20220808  
2      POLYGON ((961469.0843631719 1942206.137323901,...  20220808  
3      POLYGON ((961540.5445842013 1942201.814513462,...  20220808  
4      POLYGON ((961394.0973640293 1942081.091478479,...  20220808  
...                                                  ...       ...  
33645  POLYGON ((962668.1144004177 1955619.27428962, ...  20220830  
33646  POLYGON ((964369.3664106731 1956238.756100814,...  20220905  
33647  POLYGON ((964276.5482719766 1956221.556578399,...  20220817  
33648  POLYGON ((962431.5509421444 1956419.779457934,...  20220905  
33649  POLYGON ((964409.3988623659 1956545.591056199,...  20220830  

[33650 rows x 48 columns]

In [196]:
침수흔적도

자치구  침수수심               침수원인  침수평균해발고도      침수면적        행정코드  침수시작시간  \
0      강북구  0.02  2022년 8.8 ~17. 호우    114.53   3876.00  1130510400      18   
1      강북구  0.02  2022년 8.8 ~17. 호우    118.05   2188.00  1130510400      18   
2      강북구  0.02  2022년 8.8 ~17. 호우    129.55  45287.00  1130510400      18   
3     영등포구  0.30  2022년 8.8 ~17. 호우     10.18     53.00  1156010400      20   
3     영등포구  0.30  2022년 8.8 ~17. 호우     10.18     53.00  1156010400      20   
...    ...   ...                ...       ...       ...         ...     ...   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   
9681   양천구  0.36  2018년 8.26~9.1 호우     15.95      9.59  1147010300      20   

      침수종료시간  침수내용                      주소  침수연도 침수지역정보  \
0         23  내수침수     서울특별시 강북구 우이동 296-1  2022     주택   
1         23  내수침수     서울특별시 강북구 우이동 299-2  2022     주택   
2         23  내수침수     서울특별시 강북구 우이동 산8-15  2022     주택   
3         24  내수침수  서울특별시 영등포구 영등포동3가 13-7  2022     상가   
3         24  내수침수  서울특별시 영등포구 영등포동3가 13-7  2022     상가   
...      ...   ...                     ...   ...    ...   
9681      24  내수침수            양천구 신월동 97-4  2018    NaN   
9681      24  내수침수            양천구 신월동 97-4  2018    NaN   
9681      24  내수침수            양천구 신월동 97-4  2018    NaN   
9681      24  내수침수            양천구 신월동 97-4  2018    NaN   
9681      24  내수침수            양천구 신월동 97-4  2018    NaN   

                                                  지오메트리      침수날짜  
0     POLYGON ((956425.1271956386 1963792.043624141,...  20220808  
1     POLYGON ((956590.6408143371 1963700.11134354, ...  20220808  
2     POLYGON ((956383.2795007367 1963392.863235048,...  20220808  
3     POLYGON ((947842.214000747 1946702.945905377, ...  20220808  
3     POLYGON ((947842.214000747 1946702.945905377, ...  20220809  
...                                                 ...       ...  
9681  POLYGON ((940810.003042229 1948700.383580515, ...  20180828  
9681  POLYGON ((940810.003042229 1948700.383580515, ...  20180829  
9681  POLYGON ((940810.003042229 1948700.383580515, ...  20180830  
9681  POLYGON ((940810.003042229 1948700.383580515, ...  20180831  
9681  POLYGON ((940810.003042229 1948700.383580515, ...  20180901  

[33649 rows x 14 columns]

In [201]:
침수흔적도[침수흔적도['자치구']=='영등포구']

자치구  침수수심               침수원인  침수평균해발고도    침수면적        행정코드  침수시작시간  \
3     영등포구   0.3  2022년 8.8 ~17. 호우     10.18   53.00  1156010400      20   
3     영등포구   0.3  2022년 8.8 ~17. 호우     10.18   53.00  1156010400      20   
3     영등포구   0.3  2022년 8.8 ~17. 호우     10.18   53.00  1156010400      20   
7     영등포구   0.3  2022년 8.8 ~17. 호우     14.70  306.45  1156013300      20   
7     영등포구   0.3  2022년 8.8 ~17. 호우     14.70  306.45  1156013300      20   
...    ...   ...                ...       ...     ...         ...     ...   
4937  영등포구   0.3  2022년 8.8 ~17. 호우     12.99  130.00  1156012100      20   
4937  영등포구   0.3  2022년 8.8 ~17. 호우     12.99  130.00  1156012100      20   
4938  영등포구   0.3  2022년 8.8 ~17. 호우      9.23  326.00  1156012200      20   
4938  영등포구   0.3  2022년 8.8 ~17. 호우      9.23  326.00  1156012200      20   
4938  영등포구   0.3  2022년 8.8 ~17. 호우      9.23  326.00  1156012200      20   

      침수종료시간  침수내용                      주소  침수연도 침수지역정보  \
3         24  내수침수  서울특별시 영등포구 영등포동3가 13-7  2022     상가   
3         24  내수침수  서울특별시 영등포구 영등포동3가 13-7  2022     상가   
3         24  내수침수  서울특별시 영등포구 영등포동3가 13-7  2022     상가   
7         24  내수침수     서울특별시 영등포구대림동 951-3  2022     주택   
7         24  내수침수     서울특별시 영등포구대림동 951-3  2022     주택   
...      ...   ...                     ...   ...    ...   
4937      24  내수침수  서울특별시 영등포구 문래동3가 54-37  2022     상가   
4937      24  내수침수  서울특별시 영등포구 문래동3가 54-37  2022     상가   
4938      24  내수침수     서울특별시 영등포구 문래동4가 67  2022     상가   
4938      24  내수침수     서울특별시 영등포구 문래동4가 67  2022     상가   
4938      24  내수침수     서울특별시 영등포구 문래동4가 67  2022     상가   

                                                  지오메트리      침수날짜  
3     POLYGON ((947842.214000747 1946702.945905377, ...  20220808  
3     POLYGON ((947842.214000747 1946702.945905377, ...  20220809  
3     POLYGON ((947842.214000747 1946702.945905377, ...  20220810  
7     POLYGON ((947780.3274105673 1943886.278012975,...  20220808  
7     POLYGON ((947780.3274105673 1943886.278012975,...  20220809  
...                                                 ...       ...  
4937  POLYGON ((946758.2837477652 1946284.41428679, ...  20220809  
4937  POLYGON ((946758.2837477652 1946284.41428679, ...  20220810  
4938  POLYGON ((945964.415619435 1946224.076660675, ...  20220808  
4938  POLYGON ((945964.415619435 1946224.076660675, ...  20220809  
4938  POLYGON ((945964.415619435 1946224.076660675, ...  20220810  

[10389 rows x 14 columns]

In [202]:
하천거리

행정동  자치구      빗물펌프_거리  빗물펌프_거리(km)
0     서울특별시 종로구 사직동  종로구  4055.607378     4.055607
1     서울특별시 종로구 삼청동  종로구  4363.557106     4.363557
2     서울특별시 종로구 부암동  종로구  5619.094582     5.619095
3     서울특별시 종로구 평창동  종로구  5222.816610     5.222817
4     서울특별시 종로구 무악동  종로구  4349.727775     4.349728
..              ...  ...          ...          ...
421  서울특별시 강동구 암사1동  강동구  1350.733138     1.350733
422  서울특별시 강동구 천호2동  강동구    68.191260     0.068191
423    서울특별시 강동구 길동  강동구  2214.255829     2.214256
424  서울특별시 구로구 오류2동  구로구  2759.721782     2.759722
425   서울특별시 강동구 강일동  강동구   827.631290     0.827631

[426 rows x 4 columns]

In [204]:
하천거리.isnull().sum()

행정동            0
자치구            0
빗물펌프_거리        0
빗물펌프_거리(km)    0
dtype: int64

In [205]:
빗물펌프거리.isnull().sum()

행정동            0
자치구            0
빗물펌프_거리        0
빗물펌프_거리(km)    0
dtype: int64

0